In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
raw = pd.read_csv('C:/Users/ldmag/Downloads/website_ab_test.csv')

In [3]:
raw

Theme  Click Through Rate  Conversion Rate  Bounce Rate  \
0    Light Theme            0.054920         0.282367     0.405085   
1    Light Theme            0.113932         0.032973     0.732759   
2     Dark Theme            0.323352         0.178763     0.296543   
3    Light Theme            0.485836         0.325225     0.245001   
4    Light Theme            0.034783         0.196766     0.765100   
..           ...                 ...              ...          ...   
995   Dark Theme            0.282792         0.401605     0.200720   
996   Dark Theme            0.299917         0.026372     0.762641   
997  Light Theme            0.370254         0.019838     0.607136   
998  Light Theme            0.095815         0.137953     0.458898   
999   Dark Theme            0.342588         0.061315     0.452410   

     Scroll_Depth  Age   Location  Session_Duration Purchases Added_to_Cart  
0       72.489458   25    Chennai              1535        No           Yes  
1       61.858568   19       Pune               303        No           Yes  
2       45.737376   47    Chennai               563       Yes           Yes  
3       76.305298   58       Pune               385       Yes            No  
4       48.927407   25  New Delhi              1437        No            No  
..            ...  ...        ...               ...       ...           ...  
995     68.478822   25    Kolkata               321       Yes           Yes  
996     73.019821   38    Chennai              1635       Yes           Yes  
997     33.963298   32  Bangalore              1237        No           Yes  
998     37.429284   24    Chennai               893       Yes            No  
999     31.613326   33    Chennai               129       Yes           Yes  

[1000 rows x 10 columns]

In [4]:
map_themes = {'Light Theme': 0, 'Dark Theme': 1}
map_binary = {'No': 0, 'Yes': 1}
from sklearn.preprocessing import LabelEncoder

LE = LabelEncoder()
raw['Theme'] = LE.fit_transform(raw['Theme'])
raw['Location'] = LE.fit_transform(raw['Location'])
raw['Purchases'] = LE.fit_transform(raw['Purchases'])
raw['Added_to_Cart'] = LE.fit_transform(raw['Added_to_Cart'])

## Hypothesis 1:
Light Theme users are more likely to add something to their carts than dark users

H_a: Mu1 <= Mu2

H_0: Mu1 > Mu2

In [5]:
# split into two groups
light = raw.query('Theme == 1')
dark = raw.query('Theme == 0')

In [6]:
# randomly sample 35 points from each group with no replacement
light_sample = light.sample(300, replace=False)
dark_sample = dark.sample(300, replace=False)

In [7]:
from scipy.stats import ttest_ind

# t-test time
t,p = ttest_ind(light_sample['Added_to_Cart'], dark_sample['Added_to_Cart'], equal_var=False)

print("t-statistic: ", t)
print("p-value: ", p)

t-statistic:  1.2245674107123408
p-value:  0.2212204303031961


Given an alpha of 0.05; p > alpha (0.221) so we reject the null hypothesis. Light theme users are not more likely to add something to their carts than dark theme users.

## One way Analysis of Variance